actions/npm-basic-checks/action.yml

```diff
@@ -0,0 +1,26 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of Node.
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: "Basic Checks and Tests"
+description: "Perform basic checks and tests for an NPM project"
+inputs:
+  node-version:
+    required: true
+    description: "the node version"
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ inputs.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.node-version }}
+    - run: npm install
```

Comment:
Should we use `npm ci` here? It's meant exactly for this use case:
https://docs.npmjs.com/cli/v8/commands/npm-ci

Reply:
Bunch of our repos use yarn so it should be either configurable or action should detect it based on yarn.lock or package-lock.json files existing.

Reply:
That's fine, yarn has its own version of this: `yarn install --frozen-lockfile`

We might want to start using the name package manager throughout the org. It's simpler to not have to think about it when changing repos.

Reply:
> That's fine, yarn has its own version of this: `yarn install --frozen-lockfile`

Yeah what I'm saying is that this job should distinguish between repos that use npm and repos that use yarn

Reply:
@kwypchlo That's why I called the action `npm-basic-checks`.

Reply:
I would recommend this as a follow up.

Reply:
Switched to `npm ci`. Results of the latest run using this action here:

https://github.com/SkynetLabs/skynet-mysky/actions/runs/2158179748

Not sure what other follow-ups are needed here. Another composite action, just using `yarn` instead of `npm`?